### PR TITLE
convert tk/mem.h to memh.d

### PR DESCRIPTION
**Pull request by WalterBright**
Should use .di files, but gave up on that due to non-makefile build systems can't handle it. So call it memh.d instead.

### CODE REVIEW

**Comment by WalterBright** [comment]
@CyberShadow looks like DAutoTest is having some unrelated problem.

File: src/dmd/backend/memh.d [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * Copyright:   Copyright (C) 1985-1998 by Symantec
+ *              Copyright (C) 2000-2018 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/backend/memh.d, backend/memh.d)
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/backend/memh.d
+ */
+
+
```
Comment (jacob-carlborg) [inline src/dmd/backend/memh.d]: Some extra unnecessary whitespace here.

File: src/dmd/backend/memh.d [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+module dmd.backend.memh;
+
+
```
Comment (jacob-carlborg) [inline src/dmd/backend/memh.d]: Some extra unnecessary whitespace here.

**Review by jacob-carlborg — APPROVED** [review]

**Review by wilzbach — APPROVED** [review]
